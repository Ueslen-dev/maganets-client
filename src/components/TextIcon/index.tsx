import * as S from './styles';

type Props = {
  name: string;
  icon?: React.ReactNode;
};
const TextIcon = ({ name, icon }: Props) => (
  <S.Wrapper>
    {icon}
    <span>{name}</span>
  </S.Wrapper>
);
export default TextIcon;
