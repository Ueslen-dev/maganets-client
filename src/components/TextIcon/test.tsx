import { screen, render } from '@testing-library/react';

import TextIcon from '.';

describe('<TextIcon/>', () => {
  it('should render the heading', () => {
    const { container } = render(<TextIcon />);
    expect(
      screen.getByRole('heading', { name: /TextIcon/i })
    ).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

});
