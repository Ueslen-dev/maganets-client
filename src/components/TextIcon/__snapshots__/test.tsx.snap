// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextIcon/> should render the heading 1`] = `
<main
  class=""
>
  <h1>
    TextIcon
  </h1>
</main>
`;
